grafana_client/elements/datasource.py

```diff
@@ -301,11 +312,26 @@ def smartquery(self, datasource: Union[DatasourceIdentifier, Dict], expression:
             request_kwargs = {}
             send_request = self.client.GET
 
+        elif datasource_type in ("prometheus", "loki") and LooseVersion(self.api.version) <= VERSION_7:
+            if "queries" in request["data"] and len(request["data"]["queries"]) > 0 \
+                and "instant" in request["data"]["queries"][0] and request["data"]["queries"][0]["instant"]:
+                return self.query(
+                    datasource.get("id"),
+                    request["expr"],
+                    request["data"]["to"],
+                )
+            else:
+                return self.query_range(
+                    datasource.get("id"),
+                    request["expr"],
+                    request["data"]["from"],
+                    request["data"]["to"],
+                    request["data"]["step"],
+                )
```

Comment:
> datasource accesses change with grafana version: Prometheus in v 7 use format specified in function query() and query_range(), but now use a standard format for the queries (type or range) and results.

Hi. It looks like @PankajKhanwani is looking for this improvement around properly supporting Prometheus at GH-85, right?

Reply:
Hi,
I think so.
even if PR was built for grafana 9.2 I think that the Prometheus datasource has not changed.
Do you believe you have time to review ?
JFPIK-